gempak/ush/gfs_meta_hur.sh

```diff
@@ -1,95 +1,62 @@
-# J.W. Carr/HPC    8/05/98   Changed map to medium resolution
-# J.W. Carr/HPC    2/02/99   Changed skip to 0
-# J.W. Carr/HPC    4/12/99   Added 84-hr time step.
-# J. Carr/HPC         6/99   Added a filter to map.
-# J. Carr/HPC       2/2001   Edited to run on the IBM.
-# J. Carr/HPC       5/2001   Added a mn variable for a/b side dbnet root variable.
-# J. Carr/HPC       6/2001   Incorporated the crb metafile into this one.
-# J. Carr/HPC       6/2001   Converted to a korn shell prior to delivering script to Production.
-# J. Carr/HPC       7/2001   Submitted.
-# J. Carr/PMB      11/2004   Added a ? to all title lines. Changed contur to a 2 from a 1.
-#
 # Set up Local Variables
 #
-set -x
-#
-export PS4='hur:$SECONDS + '
-mkdir  -p -m 775 $DATA/hur
-cd $DATA/hur
-cp ${HOMEgfs}/gempak/fix/datatype.tbl datatype.tbl
+
+source "${HOMEgfs}/ush/preamble.sh"
+
+mkdir  -p -m 775 "${DATA}/hur"
+cd "${DATA}/hur" || exit 2
+cp "${HOMEgfs}/gempak/fix/datatype.tbl" datatype.tbl
 
 mdl=gfs
 MDL=GFS
 metatype="hur"
 metaname="${mdl}_${metatype}_${cyc}.meta"
 device="nc | ${metaname}"
-PDY2=$(echo ${PDY} | cut -c3-)
+
+#
+# Link data into DATA to sidestep gempak path limits
+# TODO: Replace this
+#
+export COMIN="${RUN}.${PDY}${cyc}"
+ln -sf "${COM_ATMOS_GEMPAK_1p00}" "${COMIN}"
+
 #
 # DEFINE YESTERDAY
-PDYm1=$($NDATE -24 ${PDY}${cyc} | cut -c -8)
-PDY2m1=$(echo ${PDYm1} | cut -c 3-)
+PDYm1=$(date --utc +%Y%m%d -d "${PDY} 00 - 24 hours")
 #
-if [ ${cyc} -eq 00 ] ; then
-    gdat="F000-F126-06"
-    gdatpcpn06="F006-F126-06"
-    gdatpcpn12="F012-F126-06"
-    gdatpcpn24="F024-F126-06"
-    gdatpcpn48="F048-F126-06"
-    gdatpcpn60="F060-F126-06"
-    gdatpcpn72="F072-F126-06"
-    gdatpcpn84="F084-F126-06"
-    gdatpcpn96="F096-F126-06"
-    gdatpcpn120="F120-F126-06"
-    gdatpcpn126="F126"
-    run="r"
-elif [ ${cyc} -eq 12 ] ; then
-    gdat="F000-F126-06"
-    gdatpcpn06="F006-F126-06"
-    gdatpcpn12="F012-F126-06"
-    gdatpcpn24="F024-F126-06"
-    gdatpcpn48="F048-F126-06"
-    gdatpcpn60="F060-F126-06"
-    gdatpcpn72="F072-F126-06"
-    gdatpcpn84="F084-F126-06"
-    gdatpcpn96="F096-F126-06"
-    gdatpcpn120="F120-F126-06"
-    gdatpcpn126="F126"
-    run="r"
-else
-    gdat="F000-F126-06"
-    gdatpcpn06="F006-F126-06"
-    gdatpcpn12="F012-F126-06"
-    gdatpcpn24="F024-F126-06"
-    gdatpcpn48="F048-F126-06"
-    gdatpcpn60="F060-F126-06"
-    gdatpcpn72="F072-F126-06"
-    gdatpcpn84="F084-F126-06"
-    gdatpcpn96="F096-F126-06"
-    gdatpcpn120="F120-F126-06"
-    gdatpcpn126="F126"
-    run="r"
-fi
+case ${cyc} in
+    00)
+        gdat="F000-F126-06"
+        gdatpcpn12="F012-F126-06"
+        gdatpcpn24="F024-F126-06"
+        ;;
+    12)
+        gdat="F000-F126-06"
+        gdatpcpn12="F012-F126-06"
+        gdatpcpn24="F024-F126-06"
+        ;;
+    *)
+        gdat="F000-F126-06"
+        gdatpcpn12="F012-F126-06"
+        gdatpcpn24="F024-F126-06"
+        ;;
+esac
```

Comment:
These 3 cases look the same to me. So do in the old code. Any reasons why gdatpcpn06, gdatpcpn48, gdatpcpn60, gdatpcpn72, gdatpcpn84, gdatpcpn96, gdatpcpn120, and gdatpcpn126 were taken out?

Reply:
The ones that were deleted weren't being used. I think you're right and the cyc doesn't really matter.